.github/workflows/lint.yml

```diff
@@ -0,0 +1,34 @@
+name: lint
+
+permissions:
+  id-token: write
+  contents: read
+
+on:
+  - pull_request
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  prettier:
+    runs-on: [self-hosted, ubuntu20.04-self]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 14.x
+
+      - uses: actions/cache@v2
+        name: Configure npm caching
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/workflows/lint.yml') }}
+          restore-keys: |
+            ${{ runner.os }}-npm-
+
+      - run: npm install
+
+      - name: Run prettier
+        run: npm run lint
```

Comment:
Wouldn't it be a better idea to use the prettier action? The npm caching here is also based on the lint.yml file which does not have the proper declaration for prettier so you'll always have the same npm cache

Reply:
good catch 👍